documentation/modules/cruise-control/con-optimization-goals.adoc

```diff
@@ -0,0 +1,133 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='con-optimization-goals-{context}']
+= Optimization goals overview
+
+To rebalance a Kafka cluster, Cruise Control uses optimization goals to generate xref:con-optimization-proposals-{context}[optimization proposals].  
+
+Optimization goals are targets for workload redistribution and resource utilization across a Kafka cluster.
+With a few exceptions, {ProductName} supports all the optimization goals developed in the Cruise Control project.
+These are as follows, in descending priority order:
+
+. Rack-awareness
+. Replica capacity
+. Capacity: Disk capacity, Network inbound capacity, Network outbound capacity
+//.. CPU capacity
+. Replica distribution
+. Potential network output
+. Resource distribution: Disk utilization distribution, Network inbound utilization distribution, Network outbound utilization distribution
+//.. CPU utilization distribution
+. Leader bytes-in rate distribution
+. Topic replica distribution
+. Leader replica distribution
+. Preferred leader election
+//. Intra-broker disk capacity
+//. Intra-broker disk usage distribution   
+
+For more information on each optimization goal, see link:https://github.com/linkedin/cruise-control/wiki/Pluggable-Components#goals[Goals^] in the Cruise Control Wiki.
+
+NOTE: CPU goals, intra-broker disk goals, "Write your own" goals, and Kafka assigner goals are not yet supported.
+
+As described in the following sections, you can customize the supported optimization goals by reordering them in terms of priority, and disabling goals to exclude from optimization proposal calculations.
+
+[discrete]
+== Goals configuration in {ProductName}
+
+You configure optimization goals in the `Kafka` and `KafkaRebalance` custom resources. Cruise Control has configurations for link:#master-goals[master], link:#default-goals[default], and link:#user-provided-goals[user-provided] optimization goals, as well as link:#hard-soft-goals[hard] optimization goals that must be satisfied.
+
+The following sections describe each goal configuration in more detail.
+
+[id="master-goals"]
+[discrete]
+=== Master optimization goals
```

Comment:
from where "Master" comes from?

Reply:
That's my choice of term. Otherwise we have _goals_, _default goals_, and _user-provided goals_, which are all different types of goals!

Reply:
Why not just using the CC community terms. If they are called goals, default goals, user-provided goals .. we have to take them not using something different tbh. @tomncooper @kyguy ?

Reply:
"User-provided goals" is also our choice of term.

Reply:
but user-provided is just related to the fact that the user is providing them through our technology ... the `KafkaRebalance` resource, so I think that it's right.

Reply:
personally I am happy with the "master goals list" it conveys better what they are. If a goal is not in the master goal list it is not available for any CC function.

Reply:
> but user-provided is just related to the fact that the user is providing them through our technology ... the `KafkaRebalance` resource, so I think that it's right.

Right, I see what you mean. It's more of a description than an actual term.